### PR TITLE
Survey publish: added ChainsValidationJob

### LINKED ISSUE #788: SurveyPublish: Do not allow publishing when there's at least one invalid processing chain
- [x] Add ChainsValidationJob

### PR DESCRIPTION
**Pull request by SteRiccio**
Resolves #788

### CODE REVIEW

File: server/modules/survey/service/publish/surveyPublishJob.js [hunk 1]
```diff
@@ -24,6 +25,7 @@ export default class SurveyPublishJob extends Job {
       new SurveyInfoValidationJob(),
       new CyclesDeletedCheckJob(),
       new ProcessingChainsCyclesCheckJob(),
+      new ChainsValidationJob(),
```
Comment (minotogna) [inline server/modules/survey/service/publish/surveyPublishJob.js:28]: should this go before ProcessingChainsCyclesCheckJob?

File: server/modules/survey/service/publish/jobs/chainsValidationJob.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import Job from '@server/job/job'
```
Comment (minotogna) [inline server/modules/survey/service/publish/jobs/chainsValidationJob.js:1]: relative import

File: server/modules/survey/service/publish/jobs/chainsValidationJob.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  async execute() {
+    const { surveyId } = this
+
+    const chains = await AnalysisManager.fetchChains({
+      surveyId,
+      includeStepsAndCalculations: true,
+    })
+
+    const survey = await SurveyManager.fetchSurveyById(surveyId, true)
```
Comment (minotogna) [inline server/modules/survey/service/publish/jobs/chainsValidationJob.js:25]: this can be a promise.all

File: core/i18n/resources/en.js [hunk 4]
```diff
@@ -688,6 +688,7 @@ $t(common.cantUndoWarning)`,
     CategoriesImportJob: 'Categories Import',
     CategoryImportJob: 'Category Import',
     CategoriesValidationJob: 'Categories Validation',
+    ChainsValidationJob: `Processing Chains' Validation`,
```
Comment (minotogna) [inline core/i18n/resources/en.js:691]: let's just write Processing Chains Validation

File: server/modules/survey/service/publish/surveyPublishJob.js (hunk 1 above)
Comment (SteRiccio) [reply server/modules/survey/service/publish/surveyPublishJob.js:28]: uhm, that job removes chains that won't belong to any cycles, it cleans up the chains.

File: server/modules/survey/service/publish/jobs/chainsValidationJob.js (hunk 2 above)
Comment (SteRiccio) [reply server/modules/survey/service/publish/jobs/chainsValidationJob.js:1]: right

File: server/modules/survey/service/publish/jobs/chainsValidationJob.js (hunk 3 above)
Comment (SteRiccio) [reply server/modules/survey/service/publish/jobs/chainsValidationJob.js:25]: ok

File: core/i18n/resources/en.js (hunk 4 above)
Comment (SteRiccio) [reply core/i18n/resources/en.js:691]: ok

File: server/modules/survey/service/publish/surveyPublishJob.js (hunk 1 above)
Comment (minotogna) [reply server/modules/survey/service/publish/surveyPublishJob.js:28]: right!

**Pull request merged**